var/spack/repos/builtin/packages/h5bench/package.py

```diff
@@ -70,7 +70,13 @@ def setup_build_tests(self):
         exe = os.path.basename(launcher.path)
         filename = "samples/sync-write-1d-contig-contig.json"
         filter_file(f"{exe}", f"{launcher}", filename)
-        filter_file(r"-n 1", "-n 1 --timeout 240", filename)
+        if self.spec["slurm"]:
```

Comment:
Now that I look at this again, since `slurm` is a dependency this will always be true and the test will always try to run with `srun`.

This makes it unnecessary to have `mpi_launcher()`.

On the other hand, if the goal is to support different launchers, then you probably want this line to be something like `if exe == "srun":`.

That said, I am unable to run the tests using `srun` on one of the machines at LLNL.  This occurs whether or not `slurm` is a dependency and whether line 74 replaces `mpirun` with `srun` or the full path of `srun`.